examples/react/json-based-microapp-renderer-with-css/src/components/Microapp.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright 2022 Workgrid Software
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState } from 'react'
+import { AdaptiveCardUsingHostConfigContext } from 'adaptivecards-react'
+import { useQuery, gql } from '@apollo/client'
+import Loader from './Loader'
+
+// GraphQL Query for microapp content
+const GET_MICROAPP_QUERY = gql`
+  query GetMicroapp($spaceId: ID!, $entrypoint: String!, $data: JSONObject) {
+    me {
+      space(spaceId: $spaceId) {
+        appView(entrypoint: $entrypoint, data: $data) {
+          view
+        }
+      }
+    }
+  }
+`
+
+const Microapp = ({ node }) => {
+  const [entrypoint, setEntrypoint] = useState(node.entrypoint)
+  const [queryData, setQueryData] = useState(undefined)
+
+  // GraphQL Query hook to automatically fetch and refetch data
+  const { data, loading, error } = useQuery(GET_MICROAPP_QUERY, {
+    variables: {
+      spaceId: process.env.REACT_APP_SPACE_ID,
+      entrypoint: entrypoint,
```

Comment:
I know this is just documenting the api that has been built, but calling this `entrypoint` after the first request feels really weird... just my 2 cents.

Reply:
in reality it's just a viewId - so maybe viewId makes more sense. @sjjharrington probably has some thoughts on the matter.

Reply:
Currently it maps to a featureId within Workshop however in the future it could map to something entirely different (i.e. a different source). From an external dev perspective `endpoint` seems the most self explanatory and with a view to the future gives us flexibility.

Reply:
Implementing this fix - will need to update the api schema to support the change.